protos/telemetry/telemetry.proto

```diff
@@ -577,7 +577,11 @@ enum FixType {
 message Battery {
     uint32 id = 3 [(mavsdk.options.default_value)="0"]; // Battery ID, for systems with multiple batteries
     float voltage_v = 1 [(mavsdk.options.default_value)="NaN"]; // Voltage in volts
-    float remaining_percent = 2 [(mavsdk.options.default_value)="NaN"]; // Estimated battery remaining (range: 0.0 to 1.0)
+    float remaining_percent = 2 [(mavsdk.options.default_value)="NaN"]; // Estimated battery remaining (range: 0 to 100)
+    float current_battery_a = 4 [(mavsdk.options.default_value)="NaN"]; // Battery current in Amps, NAN if autopilot does not measure the current
+    float current_consumed_ah = 5 [(mavsdk.options.default_value)="NaN"]; // Consumed charge in Amp hours, NAN if autopilot does not provide consumption estimate
```

Comment:
The battery message will be in mA hours. Same comment as for the current.

Reply:
Note also that in the new battery message there will be a status flag MAV_BATTERY_STATUS_FLAGS_CAPACITY_RELATIVE_TO_FULL that will _change/clarify the meaning of this field_. Currently the assumption is that this is the consumed current measured by a power module; you start with a full battery on power on, and this goes up, the remaining capacity goes down. In the new version, the flag will tell you that the capacity consumed (note, not "current consumed") that this value is from a smart battery and is expected to be "amount to full battery" not "based on current since vehicle powered".

It might be that you want to think about exposing an API that separates these - i.e. you might have current consumed only populated if MAV_BATTERY_STATUS_FLAGS_CAPACITY_RELATIVE_TO_FULL is not set, and a separate one for capacity in the next version.

Reply:
Consumed is generally in mAh, I can see that point. So now the question is whether we go SI as much as possible, or units that people are familiar with.